src/assets/TestFramework/DeploymentProperties.cs

```diff
@@ -0,0 +1,24 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+// <auto-generated/>
+
+#nullable disable
+
+namespace Resources.Models
+{
+    /// <summary> Deployment properties. </summary>
+    public partial class DeploymentProperties
+    {
+        /// <summary> Initializes a new instance of DeploymentProperties. </summary>
+        /// <param name="mode"> The mode that is used to deploy resources. This value can be either Incremental or Complete. In Incremental mode, resources are deployed without deleting existing resources that are not included in the template. In Complete mode, resources are deployed and existing resources in the resource group that are not included in the template are deleted. Be careful when using Complete mode as you may unintentionally delete resources. </param>
+        public DeploymentProperties(DeploymentMode mode)
+        {
+            Mode = mode;
+        }
+
+        public DeploymentMode Mode { get; }
+
+        public object Template { get; set; }
```

Comment:
You might need to update this according to some recent changes inside `Azure.ResourceManager.Resources`